modules/mcagar/src/main/ts/ephox/mcagar/loader/Loader.ts

```diff
@@ -68,9 +68,12 @@ const setup = (callbacks: Callbacks, settings: Record<string, any>, elementOpt:
   } else {
     callbacks.preInit(tinymce, settings);
 
+    const targetSettings = ShadowDom.isInShadowRoot(target) ? ({ target: target.dom() }) : ({ selector: '#' + randomId });
```

Comment:
This does nothing if you're just passing the target always below.

Reply:
Fixed.